.github/workflows/docker.yml

```diff
@@ -8,18 +8,38 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@master
-      - name: Publish to Github Docker Package Registry
-        uses: elgohr/Publish-Docker-Github-Action@2.21
+      - name: Get the version
+        id: get_version
+        run: |
+          VERSION=${GITHUB_REF#refs/tags/}
+          if [[ ${GITHUB_REF} == "refs/heads/master" ]]; then
+            VERSION=latest
+          fi
+          echo ::set-output name=VERSION::${VERSION}
+      - name: Login ghcr.io
+        uses: docker/login-action@v1
         with:
-          name: oam-dev/kubevela/vela-core
-          username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
-          registry: docker.pkg.github.com
-          tags: "latest"
-      - name: Publish to Docker Hub Registry
-        uses: elgohr/Publish-Docker-Github-Action@2.21
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.CR_PAT }}
+      - name: Login docker.io
+        uses: docker/login-action@v1
         with:
-          name: oamdev/vela-core
+          registry: docker.io
```

Comment:
Normally username/password is granted to a specified image in docker.io, why changing the image to a `registry`?

Reply:
Not changes the name.  
it split to steps. 

logins and build & push

Reply:
with the `docker/build-push-action@v2`, 

we could push with two tags.

```
ghcr.io/${{ github.repository }}/vela-core:${{ steps.get_version.outputs.VERSION }}
docker.io/oamdev/vela-core:${{ steps.get_version.outputs.VERSION }} 
```